src/Microsoft.OData.Client/EntityDescriptor.cs

```diff
@@ -772,6 +785,22 @@ internal void ClearOperationDescriptors()
             }
         }
 
+        /// <summary>
+        /// Adds a related descriptor to the list of related descriptors.
+        /// </summary>
+        /// <param name="relatedDescriptor">the related descriptor to add.</param>
+        internal void AddRelatedDescriptor(Descriptor relatedDescriptor)
+        {
+            Debug.Assert(relatedDescriptor != null, "relatedDescriptor != null");
+
+            if (this.relatedDescriptors == null)
+            {
+                this.relatedDescriptors = new List<Descriptor>();
+            }
+
+            this.relatedDescriptors.Add(relatedDescriptor);

```

Comment:
I'm concerned about this method because it's modifying the entity descriptor and it's being used in the `ProcessDescriptors` method. Which means every time a bulk update request is executed, there's a chance this descriptor list is modified. The same related descriptors might get added to the list multiple times. Are these duplicates okay?

But what's really concerning is that multiple requests could modify this list concurrently, which could lead to undefined behaviour, with side effects like lost updates.

Reply:
resolved